calico_kubernetes/calico_kubernetes.py

```diff
@@ -228,15 +299,30 @@ def _container_remove(self, hostname, orchestrator_id):
             sys.exit(1)
 
         # Remove any IP address assignments that this endpoint has
+        ip_set = set()
         for net in endpoint.ipv4_nets | endpoint.ipv6_nets:
             assert(net.size == 1)
-            self._datastore_client.unassign_address(None, net.ip)
+            ip_set.add(net.ip)
 
-        # Remove the endpoint
-        netns.remove_veth(endpoint.name)
+        logger.info("Removing IP addresses %s from endpoint %s", ip_set, endpoint.name)
+        self._datastore_client.release_ips(ip_set)
 
-        # Remove the container from the datastore.
-        self._datastore_client.remove_workload(hostname, orchestrator_id, self.docker_id)
+        # Remove the veth interface from endpoint
+        if netns.remove_veth(endpoint.name):
+            logger.info("Successfully removed veth interface from endpoint %s",
+                        endpoint.name)
+        else:
+            logger.warning("Could not remove veth interface from endpoint %s",
+                           endpoint.name)
+
+        # Remove the container/endpoint from the datastore.
+        try:
+            self._datastore_client.remove_workload(
+                hostname, orchestrator_id, self.docker_id)
```

Comment:
We've got a weird mix of passing in `hostname` vs `HOSTNAME` (same for orchestrator_id).

I'd opt for removing some of the arguments to our functions, and just always use the globally defined version when calling into libcalico.